core/src/io.jl

```diff
@@ -86,6 +86,10 @@ function load_structvector(
         nt = merge(nt, (; time = DateTime.(nt.time, dateformat"yyyy-mm-dd HH:MM:SS.s")))
     end
 
+    if hasfield(T, :control_state) && !hasfield(typeof(nt), :control_state)
```

Comment:
Is this required for backwards compatibility? Ideally we don't add hardcoded fields to this generic function. (The :time check above is already a bit ugly).

Reply:
This is indeed needed otherwise `table = StructVector{T}(nt)` below throws an error.